src/item/abstract_window_item.cpp

```diff
@@ -7,6 +7,10 @@
 AbstractWindowItem::AbstractWindowItem(QString const& typeName) :
     AbstractItem(typeName), d_ptr(new AbstractWindowItemPrivate(this))
 {
+    connect(this,&AbstractItem::nameChanged,this,[this](){
```

Comment:
We could also move the `Q_D(...)` before the lambda definition and only capture `d` instead of `this` to signal that no public member functions are used in the lambda. As soon as you add any code after the lambda you'll probably also need access to `d` outside the lambda anyway.